frontend/src/Components/CardSelector.tsx

```diff
@@ -1,39 +1,47 @@
 import classNames from 'classnames';
-import { WebSocketApi } from '../types/WebSocket';
+import { CardValue, WebSocketApi } from '../types/WebSocket';
 import classes from './CardSelector.module.css';
 import { IconCoffee } from './IconCoffee';
 import { connectToWebSocket } from './WebSocket';
 import { IconObserver } from './IconObserver';
-import { BUTTON_OBSERVER, VOTE_COFFEE, VOTE_OBSERVER } from '../constants';
+import { BUTTON_OBSERVER, VOTE_COFFEE, VOTE_NOTE_VOTED, VOTE_OBSERVER } from '../constants';
+import { JSX } from 'preact';
+
+const SPECIAL_ICONS: { [value in CardValue]?: JSX.Element } = {
+  [VOTE_OBSERVER]: <IconObserver />,
+  [VOTE_COFFEE]: <IconCoffee />,
+};
+const getCard = (
+  cardValue: CardValue,
+  isSelected: boolean,
+  isObserver: boolean,
+  socket: WebSocketApi
+) => (
+  <button
+    key={cardValue}
+    class={classNames({
+      [classes.buttonObserver]: isObserver,
+      [classes.largeCard]: !isObserver,
+      [classes.selected]: isSelected,
+    })}
+    onClick={() => socket.setVote(isSelected ? VOTE_NOTE_VOTED : cardValue)}
+  >
+    {SPECIAL_ICONS[cardValue] || cardValue}
+    {isObserver && <div class={classes.buttonObserverText}>{BUTTON_OBSERVER}</div>}
+  </button>
+);
 
 const ProtoCardSelector = ({ socket }: { socket: WebSocketApi }) => {
   const selectedCard = socket.state.votes[socket.loginData.user];
+
   return (
     <>
       <div class={classes.cardCollection}>
-        {socket.state.scale.map((cardValue) => (
-          <button
-            key={cardValue}
-            class={classNames([
-              classes.largeCard,
-              { [classes.selected]: selectedCard === cardValue },
-            ])}
-            onClick={() => socket.setVote(cardValue)}
-          >
-            {cardValue === VOTE_COFFEE ? <IconCoffee /> : cardValue}
-          </button>
-        ))}
+        {socket.state.scale.map((cardValue) =>
+          getCard(cardValue, selectedCard === cardValue, false, socket)
+        )}
       </div>
-      <button
-        class={classNames([
-          classes.buttonObserver,
-          { [classes.selected]: selectedCard === VOTE_OBSERVER },
-        ])}
-        onClick={() => socket.setVote(VOTE_OBSERVER)}
-      >
-        <IconObserver />
-        <div class={classes.buttonObserverText}>{BUTTON_OBSERVER}</div>
-      </button>
+      {getCard(VOTE_OBSERVER, selectedCard === VOTE_OBSERVER, true, socket)}
```

Comment:
i think getCard(cardValue|VOTE_OBSERVER, selectedCard === cardValue|VOTE_OBSERVER, socket.setVote) is a bit cleaner for me

Reply:
Thank you for the suggestion! I have removed the redundant boolean flag `isObserver`, since the information is already encoded in `cardValue`, and I have also replaced the `socket` object by the `setVote` callback.

The new interface of the `getCard` functions is as follows:
```
const getCard = (cardValue: CardValue, isSelected: boolean, setVote: WebSocketApi['setVote']) { ... }
```